kythe/typescript/indexer.ts

```diff
@@ -39,7 +39,8 @@ export interface VName {
 export interface Plugin {
   name: string;
   index:
-      (pathToVName: (path: string) => VName, paths: string[],
+      (pathToVName: (path: string) => VName,
+       moduleName: (path: string) => string, paths: string[],
        program: ts.Program, emit?: (obj: {}) => void) => void;
```

Comment:
Also, as this accumulates more arguments, perhaps it could itself be a named interface, like

```
interface PluginHost {
  pathToVName(...);
  moduleName(...);
}
```

Then here:

```
index(host: PluginHost): void;
```

In fact, the `Indexer` class itself might already implement this PluginHost interface, which would mean you just pass "this" to the plugin to forward this functionality to it.

Reply:
I like this much more, thanks Evan! I've implemented this.